src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs

```diff
@@ -751,6 +795,15 @@ public List<ItemBase> ListProtectedItems()
                 }).ToList();
             }
 
+            if (deleteState != 0)
+            {
+                string[] DeleteStateMap = { "TobeDeleted", "NotDeleted" };
+                itemModels = itemModels.Where(itemModel =>
+                {
+                    return ((AzureVmItem)itemModel).DeleteState == DeleteStateMap[(int)deleteState - 1];
```

Comment:
Here, you transform the enum to string  here and why not use enum or string only?